src/components/modals/AppCreator.tsx

```diff
@@ -44,20 +44,18 @@ class AppCreator extends React.Component<Props, ComponentState> {
         this.onClickCancel = this.onClickCancel.bind(this)
     }
 
-    componentDidMount() {
-        getLuisApplicationCultures()
-            .then(cultures => {
-                const cultureOptions = cultures.map<OF.IDropdownOption>(c =>
-                    ({
-                        key: c.cultureCode,
-                        text: c.cultureCode,
-                    }))
+    async componentDidMount() {
```

Comment:
I believe the change will introduce a second, extra rendering of the DOM that executes after the browser updates. I remember noting this in the ReactJS docs

Reply:
Hmm, I'm not understanding what would be different then the original. Can you explain more about which part of the change makes this behave different and what would you mean by browser update?

I think it would have always been two renders. One initially when we mount the component and another when the `cultureOptions` are loaded and `this.setState` is called.  I think any time props or state is updated `render` is called.  If the render output is the same as before there is no effect / work done by browser.  It was async in the old code and async in the new code we're just using nicer syntax now. 

Only issues I see from this change is methods new return promises instead of undefined but nothing should be depending on the return value so it shouldn't matter.

Reply:
I tested and seems to behave as expected. I would be interested to read more about it on the the docs you found though.

As far as I can tell the extra render comes from parent component loading the applications and the timings just happen to line up that it's finishes after the culture request

See: 
![image](https://user-images.githubusercontent.com/2856501/53041296-778c3100-3438-11e9-954c-fd481263d65f.png)